charts/dns/templates/external-dns-clusterrole.yaml

```diff
@@ -1,20 +1,86 @@
----
+{{- if .Values.rbac.create -}}
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
-  name: {{ include "externalDNS.fullname" . }}-{{ .Release.Name }}
+  name: {{ include "stateless-dns.fullname" . }}
 rules:
-- apiGroups: [""]
-  resources: ["services","endpoints","pods"]
-  verbs: ["get","watch","list"]
-- apiGroups: ["extensions","networking.k8s.io"]
-  resources: ["ingresses"]
-  verbs: ["get","watch","list"]
-{{- if default false .Values.externalDNS.sourceIstioObjects }}
-- apiGroups: ["networking.istio.io"]
-  resources: ["gateways", "virtualservices"]
-  verbs: ["get","watch","list"]
-{{- end }}
-- apiGroups: [""]
-  resources: ["nodes"]
-  verbs: ["get", "watch", "list"]
+{{- if or (has "node" .Values.externalDNS.sources) (has "pod" .Values.externalDNS.sources) (has "service" .Values.externalDNS.sources) (has "contour-httpproxy" .Values.externalDNS.sources) (has "gloo-proxy" .Values.externalDNS.sources) (has "openshift-route" .Values.externalDNS.sources) (has "skipper-routegroup" .Values.externalDNS.sources) }}
```

Comment:
Fair, I'll do in this PR so next one is easy to follow.

Reply:
Done.

Reply:
I ditched the idea of `include "stateless-dns.hasAny" list ("pod" "service" "etc" "etc")` because it is not so trivial.

We will need also the `sources` from the `values.yaml` so the helper would look like this:
`include "stateless-dns.hasAny" (dict "sources" .Values.externalDNS.sources "list" (list "pod" "service" "etc" "etc"))`

I think that the approach I took is way more readable that your suggestion. Let me know your thoughts anyway.